### PR TITLE
Enables deferred intent UPE for existing legacy card stores

### LINKED ISSUE #7221: Enable Deferred Intent UPE for legacy card users
### Description
Similar to https://github.com/Automattic/woocommerce-payments/issues/7003 and https://github.com/Automattic/woocommerce-payments/issues/7220, but this time instead of newly onboarded merchants/legacy UPE merchants, legacy card merchants should have the deferred UPE feature flag enabled by default.

💡 The solution for this task **should be merged between Oct 9 and Oct 27**, so that it is released as part of PaJDYF-1gY-p2

### PR DESCRIPTION
**Pull request by timur27**
Closes https://github.com/Automattic/woocommerce-payments/issues/7221

❗ **Should be merged between Oct 9 and Oct 27**. Keeping the PR status _Draft_ to avoid an accidental merge.

#### Changes proposed in this Pull Request
This PR is the third and the last step of migration to deferred intent creation UPE. Previously, we migrated split UPE stores in https://github.com/Automattic/woocommerce-payments/pull/7058 and legacy UPE stores in https://github.com/Automattic/woocommerce-payments/pull/7306. During legacy UPE stores migration, we switched to the server-side feature flag, which made the implementation in https://github.com/Automattic/woocommerce-payments/pull/7306 a bit more complex than it could have been because we needed to support both client-side and server-side stores. This PR removes all this complexity since, starting at `6.7.0`, each store should rely only on the server-side feature flag.

#### Testing instructions

Speaking high-level, we need to check that legacy card/legacy UPE/split UPE stores are now all using deferred intent creation UPE, that there's no fallback available for merchants but if we turn off the dUPE toggle for legacy card or legacy UPE, the store should roll back with corresponding notice in the Payments settings. 

> Pre-requisite: In your local server code, change https://github.com/Automattic/woocommerce-payments-server/pull/3963/files#diff-c79fe5913d8e8943eac11e58823d53e42ce3e6234736f0ed0c745fa901b53c17R1844 to compare against 6.5.0 so that your local is treated as eligible, for testing purposes

##### Legacy card stores

1. For [this configuration](https://github.com/Automattic/woocommerce-payments/assets/22319444/40d458b9-571c-4e17-a320-77d6cd3ec2c1) (no UPE is enabled), confirm that `'is_deferred_intent_creation_upe_enabled' => true,` in the account cache contents. This means that by default, even if the feature flag doesn't exist in the server's DB, deferred UPE will be treated as enabled by the server. 
2. Navigate to `Payments -> Settings` and confirm that your settings [look the following](https://github.com/Automattic/woocommerce-payments/assets/22319444/568c869f-f2b1-4d16-b26e-d251f9fa7f42), meaning no notices, banners, but UPE methods
3. Click the kebab menu on the upper right and confirm that only the _Provide feedback_ button is available, but not _Disable_, like it was previously.
4. Perform a test purchase, navigate to Stripe Dashboard, and confirm that in Payment's details, `gateway_type` equals `split_upe_with_deferred_intent_creation` in the metadata section.

##### Legacy UPE stores
* For [this configuration](https://github.com/Automattic/woocommerce-payments/assets/22319444/0344199a-2245-4c92-8ff3-392a62baf749), confirm that steps 2-4 from the previous paragraph are leading to the same result.
* In your local server environment, run `wp wcpay disable_dupe acct_{YOUR_ACCOUNT_ID}`. With this action, we're testing the process of disabling dUPE for merchants if needed. Refresh the cache afterward.
* Navigate to payments settings and confirm that your settings [look the following](https://github.com/Automattic/woocommerce-payments/assets/22319444/d042c04a-0813-4eee-bc1a-dadfc387ed49). The text is being currently aligned with the Design Team and I'll update the text with the final version once confirmed. 
* Perform a purchase, go to Stripe Dashboard to confirm `gateway_type = legacy_upe` in the payment's details.
* Enable deferred UPE flag back with `wp wcpay enable_dupe acct_{YOUR_ACCOUNT_ID}` and refresh the cache
* Navigate to _Payments -> Settings_ and confirm that the notice shown previously disappeared. This means your store now uses deferred UPE again.

##### Split UPE stores
* For [this configuration](https://github.com/Automattic/woocommerce-payments/assets/22319444/f98b17c9-4780-419f-847c-4e352d0bea2c), confirm that steps 2-4 from the previous paragraph are leading to the same result.
* We're not testing split UPE roll back because split UPE stores were migrated some time ago, and no roll back for them is considered.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7328` or branch name `feat/enable-dupe-for-legacy-card` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=feat/enable-dupe-for-legacy-card) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: d84facad9802e589d5eda07704493ebdb6c0a320
- Build time: 2023-10-16 14:11:40 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/payment-methods/index.js [hunk 1]
```diff
@@ -302,6 +242,30 @@ const PaymentMethods = () => {
...
+				{ ! isUpeEnabled && (
+					<CardHeader className="payment-methods__header">
+						<InlineNotice
+							icon
+							status="warning"
+							isDismissible={ false }
+						>
+							{ interpolateComponents( {
+								mixedString: __(
+									'You were rolled back from dUPE to legacy card. Stay tuned on next actions.' +
```
Comment (timur27) [inline client/payment-methods/index.js]: The wording is currently being aligned with the Design Team. paJDYF-9xf-p2#comment-20974 I'll update the text once confirmed.

File: includes/class-wc-payments-features.php [hunk 2]
```diff
@@ -90,39 +75,6 @@ public static function is_upe_split_enabled() {
...
-
-	/**
-	 * Checks if the store has UPE enabled and the server-side feature flag is enabled.
-	 * This is applicable to:
-	 * * Split UPE stores starting 6.4.0
-	 * * Legacy UPE stores starting 6.6.0
-	 */
-	private static function has_store_enabled_upe_and_dupe_server_flag() {
-		return ( self::is_upe_split_enabled() || self::is_upe_legacy_enabled() ) && self::is_deferred_upe_server_flag_enabled();
-	}
```
Comment (timur27) [inline includes/class-wc-payments-features.php:119]: This complexity was introduced in https://github.com/Automattic/woocommerce-payments/pull/7306 and was needed for version `6.6.x`. Starting `6.7.0`, every store relies on the server-side flag only, thus this logic can be removed.

**Review by gpressutto5 — APPROVED** [review]
The code looks good and all scenarios are running perfectly!

**Comment by timur27** [comment]
01ec6bbb938326811b912455f87592ba8c391531 has been added after the review, to fix e2e tests which were un-skipped in https://github.com/Automattic/woocommerce-payments/pull/7334. 

@mgascam will review this single commit that fixes the E2E test.

**Comment by mgascam** [comment]
> [01ec6bb](https://github.com/Automattic/woocommerce-payments/commit/01ec6bbb938326811b912455f87592ba8c391531) has been added after the review, to fix e2e tests which were un-skipped in #7334.
> 
> @mgascam will review this single commit that fixes the E2E test.

@timur27 code looks good to me and the e2e test is green when I run it locally.

**Review by mgascam — APPROVED** [review]
LGTM!

**Pull request merged**